api/src/main/java/org/openmrs/module/fhir2/api/FhirAllergyIntoleranceService.java

```diff
@@ -11,9 +11,18 @@
 
 import javax.validation.constraints.NotNull;
 
+import java.util.Collection;
+
+import ca.uhn.fhir.rest.param.ReferenceParam;
+import ca.uhn.fhir.rest.param.StringOrListParam;
+import ca.uhn.fhir.rest.param.TokenOrListParam;
 import org.hl7.fhir.r4.model.AllergyIntolerance;
 
 public interface FhirAllergyIntoleranceService {
 	
 	AllergyIntolerance getAllergyIntoleranceByUuid(@NotNull String uuid);
+	
+	Collection<AllergyIntolerance> searchForAllergies(ReferenceParam patientReference, StringOrListParam category,
+	        TokenOrListParam allergen, TokenOrListParam severity, TokenOrListParam manifestationCode,
+	        StringOrListParam clinicalStatus);
```

Comment:
`clinicalStatus` should be a `TokenOrListParam`. The definitions for the values taken by these parameters can be found [here](https://www.hl7.org/fhir/allergyintolerance.html#search). Note that the difference between a `TokenParam` and a `StringParam` is that the `TokenParam` can contain a `system` value indicating what value-set the value it encodes represents.